gh-116126: Implement PEP 696 (#116129)
python/cpython: Doc/library/ast.rst

:ref:`Type parameters <type-params>` can exist on classes, functions, and type
aliases.

.. class:: TypeVar(name, bound, default_value)

   A :class:`typing.TypeVar`. *name* is the name of the type variable.
   *bound* is the bound or constraints, if any. If *bound* is a :class:`Tuple`,
   it represents constraints; otherwise it represents the bound. *default_value*

>    is the default value, if any.

Reviewer: It might be worth spelling this out a bit more, since this is now different to what happens at runtime. (I think that's the correct choice; the distinction between `None` for no default and `ast.Constant(value=None)` is quite clean in the AST.)

```suggestion
   is the default value; if the :class:`!TypeVar` has no default, this 
   attribute will be set to ``None``.
```

   .. doctest::

        >>> print(ast.dump(ast.parse("type Alias[T: int = bool] = list[T]"), indent=4))
        Module(
            body=[
                TypeAlias(
                    name=Name(id='Alias', ctx=Store()),
                    type_params=[
                        TypeVar(
                            name='T',
                            bound=Name(id='int', ctx=Load()),
                            default_value=Name(id='bool', ctx=Load()))],
                    value=Subscript(
                        value=Name(id='list', ctx=Load()),
                        slice=Name(id='T', ctx=Load()),
                        ctx=Load()))])

   .. versionadded:: 3.12

   .. versionchanged:: 3.13
      Added the *default_value* parameter.

.. class:: ParamSpec(name, default_value)

   A :class:`typing.ParamSpec`. *name* is the name of the parameter specification.

>    *default_value* is the default value, if any.

Reviewer: ```suggestion
   *default_value* is the default value; if the :class:`!ParamSpec` has no default,
   this attribute will be set to ``None``.
```

   .. versionadded:: 3.12

   .. versionchanged:: 3.13
      Added the *default_value* parameter.

.. class:: TypeVarTuple(name, default_value)

   A :class:`typing.TypeVarTuple`. *name* is the name of the type variable tuple.

>    *default_value* is the default value, if any.

Reviewer: ```suggestion
   *default_value* is the default value; if the :class:`!TypeVarTuple` has no
   default, this attribute will be set to ``None``.
```

   .. versionadded:: 3.12

   .. versionchanged:: 3.13

>       Added ``default_value``.

Reviewer: ```suggestion
      Added the *default_value* parameter.
```

   .. versionadded:: 3.12

   .. versionchanged:: 3.13

>       Added ``default_value``.

Reviewer: ```suggestion
      Added the *default_value* parameter.
```

.. class:: TypeVarTuple(name)

   A :class:`typing.TypeVarTuple`. ``name`` is the name of the type variable tuple.

>    ``default_value`` is the default value, if any.

Reviewer 2: You need to update the `.. class` def to include the *default\_value* parameter.

Reviewer 2: (Also, we should create a follow-up PR to make sure all parameter names are marked up according to the devguide.)

Author: Filed https://github.com/python/cpython/issues/118181; I'll fix the ones touched by this PR though.

   .. versionadded:: 3.12

   .. versionchanged:: 3.13

>       Added ``default_value``.

Reviewer: ```suggestion
      Added the *default_value* parameter.
```
